packages/sdk/src/store/base.ts

```diff
@@ -0,0 +1,58 @@
+type CancelFn = () => void;
+
+type Subscriber<T> = (value: T) => void | CancelFn;
+
+export interface Store<T> {
+  set: (val: T) => void;
+  read: () => T;
+  readInitial: () => T;
+  subscribe: (sub: Subscriber<T>) => () => void;
+}
+
+/**
+ * Creates a new Suspense ready Store
+ */
+export const createStore = <T>(
+  initialValue: T,
+): Store<T> => {
+  const subscribers = new Set<Subscriber<T>>();
+  const cancelations = new Set<CancelFn>();
+
+  let value: T = initialValue;
+
+  const read = () => value;
+
+  const readInitial = () => initialValue;
+
+  const set = (val: T) => {
+    value = val;
+
+    // Cancel previous, dangling operations
+    cancelations.forEach((cancel) => cancel());
+    cancelations.clear();
+
+    // Broadcast subscribers
+    subscribers.forEach((sub) => {
```

Comment:
Should we really notify subscribers even if value=val?

Reply:
I found it simpler to remove any ifs from inside the store. The idea is that if you want to use `Object.is` as a comparison method, you could do:
 ```
 if (store.read() !== val) {
      store.set()
 }```